src/detect-filesize.c

```diff
@@ -158,35 +158,37 @@ static DetectFilesizeData *DetectFilesizeParse (const char *str)
 
     SCLogDebug("ret %d", ret);
 
-    res = pcre_get_substring((char *)str, ov, MAX_SUBSTRINGS, 1, &str_ptr);
+    res = SC_pcre2_substring_get(parse_regex.match, 1, (PCRE2_UCHAR8 **)&str_ptr, &pcre2_len);
```

Comment:
Why `SC_pcre2_substring_get` here but `pcre2_substring_get_bynumber` in other places

Reply:
This is a good question @jlucovsky

As you can see in detect-parse.c, the only difference is that `SC_pcre2_substring_get` transformes the error `PCRE2_ERROR_UNSET` into no error and returning a NULL pointer with length 0 (adding `*bufflen = 0;` into it by the way)

This is a difference between PCRE2 and PCRE1 as PCRE1 used to return a NULL pointer, length 0 and no error for when PCRE2 returns error `PCRE2_ERROR_UNSET`

So, what do you think ?
Should we generalize `SC_pcre2_substring_get` instead of `pcre2_substring_get_bynumber` ?